cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -230,151 +204,115 @@ cros_gralloc_driver::cros_gralloc_driver()
 		}
 
 		// hit any of undesired render node
-		if (j < ARRAY_SIZE(undesired))
+		if (j < ARRAY_SIZE(undesired)) {
+			drmFreeVersion(version);
+			close(fd);
 			continue;
-
-		if (!strcmp(version->name, "virtio_gpu")) {
-			if (virtio_node_idx == -1)
-				virtio_node_idx = availabe_node;
-			else if (ivshm_node_idx == -1)
-				ivshm_node_idx = availabe_node;
 		}
 
-		if (!strcmp(version->name, "i915")) {
-			// Prefer i915 for performance consideration.
-			//
-			// TODO: We might have multiple i915 devices in the system and in
-			// this case we are effectively using the one with largest card id
-			// which is normally the dGPU VF or dGPU passed through device.
-			renderer_idx = availabe_node;
-			// Use first available i915 node for video bo alloc
-			if (video_idx == -1)
-				video_idx = availabe_node;
+		if (fallback_fd == -1)
```

Comment:
According to the code logic afterward, 'fallback_fd' should be set like:
"
if (gpu_grp_type_idx == -1)
    fallback_fd = fd;
"
Right?